src/EventTickets/Routes/DeleteEventTicketType.php

```diff
@@ -30,14 +34,16 @@ public function registerRoute()
                 [
                     'methods' => WP_REST_Server::DELETABLE,
                     'callback' => [$this, 'handleRequest'],
-                    'permission_callback' => [$this, 'permissionsCheck'],
+                    'permission_callback' => function () {
+                        return current_user_can('delete_give_payments');
+                    },
                 ],
                 'args' => [
                     'ticket_type_id' => [
                         'type' => 'integer',
                         'sanitize_callback' => 'absint',
-                        'validate_callback' => function ($eventId) {
-                            return EventTicketType::find($eventId);
+                        'validate_callback' => function ($ticketTypeId) {
+                            return give(EventTicketTypeRepository::class)->queryById($ticketTypeId)->count() > 0;
```

Comment:
What is the benefit of validating against the count, as opposed to `EventTicketType|null`? Using the facade, if the entity does not exist then it returns null, which should fail the validation.

Also, if we are to validate against the count, comparing against zero is not necessary, since a positive integer is already `truthy` and zero is `falsey`.

Reply:
I only noticed the issue when writing unit tests. The validation callback was returning `null` for invalid ticket type ids without halting execution. The reason is that the `validate_callback` checks for a false return from that method, and `null` is not considered false.

![CleanShot 2024-03-26 at 17 36 03](https://github.com/impress-org/givewp/assets/3921017/f7aac397-1c1a-45f8-8e6e-882db45219ee)

A solution would be check against `null` like in the snippet bellow. What do you think?
```php
'validate_callback' => function ($ticketTypeId) {
    return EventTicketType::find($ticketTypeId) !== null;
},

Reply:
Yeah, I think it would be simpler to check against `null` and leverage our facade.

As an aside, we might consider a `::findOrFail($id)` method in the future (another good Friday project).

Reply:
Changed all to be like `Model::find($entityId) !== null` on ef40760a21e94d8b887460706f6b97d63be4915a